src/common/imagpng.cpp

```diff
@@ -342,6 +343,35 @@ wxPNGImageData::DoLoadPNGFile(wxImage* image, wxPNGInfoStruct& wxinfo)
         return;
 
     png_read_image( png_ptr, lines );
+
+    // load "Description" text chunk
+    png_textp text_ptr;
+    int num_comments = png_get_text( png_ptr, info_ptr, &text_ptr, nullptr );
+    for (int i = 0; i < num_comments; ++i)
+    {
+        if (text_ptr[i].compression == PNG_TEXT_COMPRESSION_NONE) // Latin-1 encoding
+        {
+            wxString key = wxString::From8BitData(text_ptr[i].key);
+            if (key == wxIMAGE_OPTION_PNG_DESCRIPTION_KEY)
+            {
+                wxString text = wxString::From8BitData(text_ptr[i].text);
+                image->SetOption(wxIMAGE_OPTION_PNG_DESCRIPTION, text);
+                break;
+            }
+        }
+        else if (text_ptr[i].compression == PNG_ITXT_COMPRESSION_NONE) // UTF-8 encoding
+        {
+            wxString key = wxString::FromUTF8(text_ptr[i].key);
```

Comment:
Sorry, this is wrong, PNG specification says:

> The keyword must be at least one character and less than 80 characters long. Keywords are always interpreted according to the ISO/IEC 8859-1 (Latin-1) character set [[ISO/IEC-8859-1](https://www.libpng.org/pub/png/spec/1.2/PNG-References.html#B.I-8859-1)]. They must contain only printable Latin-1 characters and spaces; that is, only character codes 32-126 and 161-255 decimal are allowed.

So the key should always be passed to `From8BitData()`.